src/app_service/service/collectible/service.nim

```diff
@@ -47,37 +59,74 @@ QtObject:
-      self.events.emit(GetCollections, GetCollectionsArgs(collections: collections))
-    except:
-      self.events.emit(GetCollections, GetCollectionsArgs())
+      if (responseObj.kind == JObject):
+        let chainIdJson = responseObj["chainId"]
+        let addressJson = responseObj["address"]
+
+        let validAccount = (chainIdJson.kind == JInt and 
+          addressJson.kind == JString)
+        if (validAccount):
+          data.chainId = chainIdJson.getInt()
+          data.address = addressJson.getStr()
+
+          let collectionsJson = responseObj["collections"]
+          if (collectionsJson.kind == JArray):
+            data.collections = map(collectionsJson.getElems(), proc(x: JsonNode): CollectionDto = x.toCollectionDto())
+    except Exception as e:
+      let errDesription = e.msg
+      error "error onGetCollections: ", errDesription
+    self.events.emit(GetCollections, data)
 
   proc getCollectionsAsync*(self: Service, address: string) =
     let arg = GetCollectionsTaskArg(
       tptr: cast[ByteAddress](getCollectionsTaskArg),
       vptr: cast[ByteAddress](self.vptr),
       slot: "onGetCollections",
-      chainId: self.networkService.getNetworkForCollectibles().chainId,
+      chainId: self.getNetwork().chainId,
       address: address,
     )
     self.threadpool.start(arg)
-
-  proc getNetwork*(self: Service): NetworkDto =
-    return self.networkService.getNetworkForCollectibles()
   
-  proc getCollectibles*(self: Service, address: string, collectionSlug: string): seq[CollectibleDto] =
+  proc onGetCollectibles*(self: Service, response: string) {.slot.} =
+    var data = GetCollectiblesArgs()
     try:
-      let chainId = self.getNetwork().chainId
-      let response = backend.getOpenseaAssetsByOwnerAndCollection(chainId, address, collectionSlug, limit)
-      return map(response.result.getElems(), proc(x: JsonNode): CollectibleDto = x.toCollectibleDto())
+      let responseObj = response.parseJson
+      if (responseObj.kind == JObject):
+        let chainIdJson = responseObj["chainId"]
+        let addressJson = responseObj["address"]
+        let collectionSlugJson = responseObj["collectionSlug"]
+        
+        let validCollection = (chainIdJson.kind == JInt and 
+          addressJson.kind == JString and 
+          collectionSlugJson.kind == JString)
+        if (validCollection):
+          data.chainId = chainIdJson.getInt()
+          data.address = addressJson.getStr()
+          data.collectionSlug = collectionSlugJson.getStr()
+
+          let collectiblesJson = responseObj["collectibles"]
+          if (collectiblesJson.kind == JArray):
+            data.collectibles = map(collectiblesJson.getElems(), proc(x: JsonNode): CollectibleDto = x.toCollectibleDto())
     except Exception as e:
       let errDesription = e.msg
-      error "error: ", errDesription
-      return
+      error "error onGetCollectibles: ", errDesription
+    self.events.emit(GetCollectibles, data)
```

Comment:
@dlipicar please rename this signal to `collectiblesFetched` or some name by your choice which will describe what we're doing, this way is a bit confusing for one who connects to it. I would expect that I have to get/fetch them from somewhere if I would receive it.

Reply:
I did exactly that on a separate PR :D https://github.com/status-im/status-desktop/issues/8407
In the new PR, it gets renamed to "SIGNAL_COLLECTIBLES_UPDATED", here I only wanted to be consistent with the signal name we have for Collections. 

I'll make it ready for review when this one gets merged.